docs/frontend-system/building-plugins/migrating.md

```diff
@@ -0,0 +1,189 @@
+import { workApiRef } from '@internal/plugin-example-react';
+import { WorkImpl } from './WorkImpl';
+
+const exampleWorkApi = createApiFactory({
+  api: workApiRef,
+  deps: { storageApi: storageApiRef },
+  factory: ({ storageApi }) => new WorkImpl({ storageApi }),
+});
+
+/** @public */
+export const catalogPlugin = createPlugin({
+  id: 'example',
+  apis: [exampleWorkApi],
+});
+```
+
+The major changes we'll make are
+
+- Optionally change the old imports to the new package as per the top section of this guide
+- Wrap the existing API factory in a `createApiExtension`
+- Change to the new version of `createPlugin` which exports this extension
+- Change the plugin export to be the default instead
+
+The end result, after simplifying imports and cleaning up a bit, might look like this:
+
+```tsx title="in @internal/plugin-example"
+import {
+  storageApiRef,
+  createPlugin,
+  createApiFactory,
+  createApiExtension,
+} from '@backstage/frontend-plugin-api';
+import { workApiRef } from '@internal/plugin-example-react';
+import { WorkImpl } from './WorkImpl';
+
+const exampleWorkApi = createApiExtension({
+  factory: createApiFactory({
+    api: workApiRef,
+    deps: { storageApi: storageApiRef },
+    factory: ({ storageApi }) => new WorkImpl({ storageApi }),
+  }),
+});
+```
+
+### Further work
+
+Since utility APIs are now complete extensions, you may want to take a bigger look at how they used to be used, and what the new frontend system offers. You may for example consider [adding configurability or inputs](../utility-apis/02-creating.md) to your API, if that makes sense for your current application.
+
+## Plugin
+
+In the legacy frontend system a plugin was defined in its own `plugin.ts` file as following:
+
+```ts title="my-plugin/src/plugin.ts"
+  import { createPlugin } from '@backstage/core-plugin-api';
+
+  export const myPlugin = createPlugin({
+    id: 'my-plugin',
+    apis: [exampleWorkApi],
+    routes: {
+      ...
+    },
+    externalRoutes: {
+      ...
+    },
+  });
+```
+
+In order to migrate the actual definition of the plugin you need to recreate the plugin using the new `createPlugin` utility exported by `@backstage/frontend-plugin-api`.
+The new `createPlugin` function doesn't accept apis anymore as apis are now extensions.
+
+```ts title="my-plugin/src/index.ts"
+  import { createPlugin } from '@backstage/frontend-plugin-api';
+
+  export default createPlugin({
+    id: 'my-plugin',
+    // bind all the extensions to the plugin
+    /* highlight-next-line */
+    extensions: [homePage, exampleWorkApi],
+    // convert old route refs to the new system
+    /* highlight-next-line */
+    routes: convertLegacyRouteRefs({
+      ...
+    }),
+    /* highlight-next-line */
+    externalRoutes: convertLegacyRouteRefs({
+      ...
+    }),
+  });
+```
+
+The code above binds all the extensions to the plugin. _Important_: Make sure to export the plugin as default export of your package in `src/index.ts`.
```

Comment:
Hmm, let's not encourage users to switch out the plugin completely at this point, and instead document how to create a separate /alpha entry-point. We'll of course need to update that in the future, but it's the most useful thing to document right now since that's what you'll want to do.

Not a lot of changes needed for that though really, just need to lead with instructions to create a separate entry point. The same before/after comparisons and import switches still apply anyway.